laws/shared/src/test/scala/algebra/laws/LawTests.scala

```diff
@@ -143,5 +143,32 @@ trait LawTestsBase extends FunSuite with Discipline {
     implicit val longSemilattice: Semilattice[Long] = LongMinMaxLattice.joinSemilattice
     laws[LatticeLaws, (Set[Int], Long)].check(_.semilattice(lexicographicSemilattice))
   }
+
+  {
+    // This "arbitrary int order" isn't that arbitrary.
+    // I think this could be improved once we have a version of Scalacheck with Cogen.
+    implicit val arbOrderInt: Arbitrary[Order[Int]] = Arbitrary(
+      Gen.oneOf(
+        Gen.const(Order[Int]),
+        Gen.const(Order[Int].reverse)))
+
+    // This is a hack to fake an `Eq` instance for an `Order`.
```

Comment:
It's really just about making the comparison feasible (something smaller would probably be even better). We'd still have too many permutations to expect equality to be common at all, but we're really just more concerned about the instances that aren't equal anyways, for the counter examples (eg `(ab)c != a(bc)`).